main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"syscall"
+
+	logrus "github.com/sirupsen/logrus"
+)
+
+// AppPath returns the application path based on the OS.
+func AppPath(goos, homeDir string) string {
+	// Determine the application path based on the operating system.
+	switch goos {
+	case "windows":
+		// On Windows, use the LOCALAPPDATA environment variable.
+		return filepath.Join(os.Getenv("LOCALAPPDATA"), DefaultAppName)
+
+	case "darwin":
+		// On macOS, place the application data in the
+		// "Application Support" directory.
+		return filepath.Join(
+			homeDir, "Library", "Application Support", DefaultAppName,
+		)
+
+	case "plan9":
+		// On Plan 9, use the home directory with the application name
+		// in lowercase.
+		return filepath.Join(homeDir, strings.ToLower(DefaultAppName))
+
+	// POSIX (Linux, etc.)
+	default:
+		// On other POSIX systems (e.g., Linux), use a hidden directory
+		// in the home directory.
+		appName := fmt.Sprintf(".%s", strings.ToLower(DefaultAppName))
+		return filepath.Join(homeDir, appName)
+	}
+}
+
+// EnsureAppPathExists ensures the application directory exists.
+func EnsureAppPathExists(path string) error {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		err := os.MkdirAll(path, AppDirPermissions)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func main() {
+	// Get the user home directory depending on the OS.
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		logrus.Fatalf("Failed to get user home directory: %v", err)
+	}
+	// Get the app path directory.
+	appPath := AppPath(runtime.GOOS, homeDir)
+
+	// Create App Path if it doesn't exist.
+	err = EnsureAppPathExists(appPath)
+	if err != nil {
+		logrus.Fatalf("Failed to create app directory: %v",
+			err)
+	}
+
+	// Initiate Config.
+	config, err := initConfig(appPath, DefaultConfigFilename)
```

Comment:
Does viper also have cmd line options ?

Reply:
Let me check this and let you know...

Reply:
Okay, Viper can bind command-line flags (e.g., from [Cobra](https://github.com/spf13/cobra)) to configuration values. Should I implement this command-line options feature ?

Reply:
not for now, but let's track it as an issue maybe for the future ?